runtime/compiler/env/VMJ9.cpp

```diff
@@ -5308,7 +5310,13 @@ TR_J9VMBase::getStringFieldByName(TR::Compilation * comp, TR::SymbolReference *
    TR::Symbol::RecognizedField   field = fieldRef->getSymbol()->getRecognizedField();
 
    if (field == TR::Symbol::Java_lang_String_count)
+      {
```

Comment:
Should the entire if be removed when not java 8? I gather the field only exists there, so there should be no possibility of getting the Symbol (in fact, should it not be ifdeffed out?)?

Reply:
I'd like to defer that kind of cleanup for separate pull requests. It's not clear how far the effects will spread when we start pulling threads like that.

Reply:
Fair enough.